pkg/gateway/model_build_targetgroup.go

```diff
@@ -172,65 +164,82 @@ func (t *latticeServiceModelBuildTask) buildTargets(ctx context.Context) error {
 	return nil
 }
 
-// TODO have a same BuildTargetGroup for both targetGroupModelBuildTask, latticeServiceModelBuildTask
-// Build target group for K8S serviceexport object
-func (t *targetGroupModelBuildTask) BuildTargetGroup(ctx context.Context) error {
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExport(ctx context.Context) error {
 
 	tgName := latticestore.TargetGroupName(t.serviceExport.Name, t.serviceExport.Namespace)
+	var tg *latticemodel.TargetGroup
+	var err error
+	if t.serviceExport.DeletionTimestamp.IsZero() {
+		// build TargetGroup Model for serviceExport creation
+		tg, err = t.buildTargetGroupForServiceExportCreation(ctx, tgName)
+	} else {
+		// build TargetGroup Model for serviceExport deletion
+		tg, err = t.buildTargetGroupForServiceExportDeletion(ctx, tgName)
+	}
+	if err != nil {
+		return err
+	}
+	t.tgByResID[tgName] = tg
+	t.targetGroup = tg
+	return nil
+}
 
+func (t *targetGroupModelBuildTask) buildTargetGroupForServiceExportCreation(ctx context.Context, targetGroupName string) (*latticemodel.TargetGroup, error) {
 	svc := &corev1.Service{}
 	if err := t.Client.Get(ctx, k8s.NamespacedName(t.serviceExport), svc); err != nil {
 		glog.V(6).Infof("Error finding corresponding service %v error :%v \n", k8s.NamespacedName(t.serviceExport), err)
 		// mark there is no serviceexport dependence on the TG
-		t.Datastore.SetTargetGroupByServiceExport(tgName, false, false)
-		return err
+		t.Datastore.SetTargetGroupByServiceExport(targetGroupName, false, false)
+		return nil, err
 	}
-
-	//if t.serviceExport.
-	tgSpec := latticemodel.TargetGroupSpec{
-		Name: tgName,
+	tg := latticemodel.NewTargetGroup(t.stack, targetGroupName, latticemodel.TargetGroupSpec{
+		Name: targetGroupName,
 		Type: latticemodel.TargetGroupTypeIP,
 		Config: latticemodel.TargetGroupConfig{
 			VpcID: config.VpcID,
 			// Fill in default HTTP port as we are using target port anyway.
-			Port:                80,
-			IsServiceImport:     false,
-			IsServiceExport:     true,
-			K8SServiceName:      t.serviceExport.Name,
-			K8SServiceNamespace: t.serviceExport.Namespace,
-			Protocol:            "HTTP",
-			ProtocolVersion:     vpclattice.TargetGroupProtocolVersionHttp1,
+			Port:                  80,
+			IsServiceImport:       false,
+			IsServiceExport:       true,
+			K8SServiceName:        t.serviceExport.Name,
+			K8SServiceNamespace:   t.serviceExport.Namespace,
+			K8SHTTPRouteName:      "", // When build TG model for serviceExport, the httproutename should be ""
+			K8SHTTPRouteNamespace: "",
```

Comment:
Explicitly show that  When build TG model for serviceExport, the httproutename should be "",  this is imporment, if we are not aware that it can cause some bug that hard to figure out. I will try to imporve our code to remove such "implicit convention"